k4FWCore/python/k4FWCore/utils.py

```diff
@@ -18,10 +18,35 @@
 # limitations under the License.
 #
 import os
+import re
 from io import TextIOWrapper
 from typing import Union
 
 
+def check_wrong_imports(code: str) -> None:
+    """Check for wrong imports in the given code.
+
+    This function checks the given code for any imports of IOSvc or ApplicationMgr
+    from Configurables instead of k4FWCore. If such an import is found, an ImportError
+    is raised.
+
+    Args:
+        code (str): The code to check for wrong imports.
+
+    Raises:
+        R
+
+    """
+    iosvc_regex = re.compile(
+        r"^\s*from\s+(Configurables|k4FWCore)\s+import\s+\(?.*IOSvc.*\)?", re.MULTILINE
+    )
+    regex = re.compile(
+        r"^\s*from\s+Configurables\s+import\s+\(?.*(ApplicationMgr|IOSvc).*\)?", re.MULTILINE
+    )
+    if re.search(iosvc_regex, code) and re.search(regex, code):
```

Comment:
Can you comment the idea for this check, it isn't obvious to me why you need both iosvc_regex and regex to be found.

Reply:
Comment added